Separate camera code from SDL code

smcameron opened the pull request:
Hi,

I found your web camera code to be quite helpful.  I forked your repo awhile ago, and made some changes to separate the camera code from the SDL code, and make like a little camera library.

I also changed the formatting (use tabs, not space, more like linux kernel code) and added a Makefile and got rid of the build shell script.

If you would like to merge these changes, feel free, but do not feel any obligation to even look at this (I realize you might not like all the changes I've made, and also that you haven't touched this code in a long time and might have lost interest.)

Mostly just wanted to say thanks, I found your code to be quite helpful.

-- steve

smcameron committed:
Unglobalize dev_name

Signed-off-by: Stephen M. Cameron <stephenmcameron@gmail.com>

smcameron committed:
Remove build.sh script

Signed-off-by: Stephen M. Cameron <stephenmcameron@gmail.com>

smcameron committed:
Fix some warnings

Signed-off-by: Stephen M. Cameron <stephenmcameron@gmail.com>

smcameron committed:
Fix formatting

Signed-off-by: Stephen M. Cameron <stephenmcameron@gmail.com>

smcameron committed:
Add Makefile

Signed-off-by: Stephen M. Cameron <stephenmcameron@gmail.com>

smcameron committed:
Split video capture from video display

Signed-off-by: Stephen M. Cameron <stephenmcameron@gmail.com>

smcameron committed:
Remove executable from git

Signed-off-by: Stephen M. Cameron <stephenmcameron@gmail.com>

smcameron committed:
Remove io_method typedef

Just use "enum camera_io_method"

Signed-off-by: Stephen M. Cameron <stephenmcameron@gmail.com>

smcameron committed:
Unglobalize int fd

Signed-off-by: Stephen M. Cameron <stephenmcameron@gmail.com>

smcameron committed:
Use sysconf instead of getpagesize()

Signed-off-by: Stephen M. Cameron <stephenmcameron@gmail.com>

smcameron committed:
Move render cookie from camera_open_device() to camera_start_capture()

Signed-off-by: Stephen M. Cameron <stephenmcameron@gmail.com>

smcameron committed:
Move io method parameter to camera_init_device()

Signed-off-by: Stephen M. Cameron <stephenmcameron@gmail.com>

djtrance commented:
> Hi,
> 
> I found your web camera code to be quite helpful. I forked your repo awhile ago, and made some changes to separate the camera code from the SDL code, and make like a little camera library.
> 
> I also changed the formatting (use tabs, not space, more like linux kernel code) and added a Makefile and got rid of the build shell script.
> 
> If you would like to merge these changes, feel free, but do not feel any obligation to even look at this (I realize you might not like all the changes I've made, and also that you haven't touched this code in a long time and might have lost interest.)
> 
> Mostly just wanted to say thanks, I found your code to be quite helpful.
> 
> -- steve

Hi steve, I just saw your message, I don't know why GitHub don't tell me about it. I will check your changes, thanks for your contributions

smcameron commented:
No problem.

BTW, on my older linux distro (I forget which one, some flavor of Ubuntu)
this all worked fine, but lately (linux Mint 20.3 based on Ubuntu focal), I
get this (which I believe is not a problem with your software, but is
rather a problem with the linux kernel header files):

$ make
cc -Wall -Wextra --pedantic -std=c99 -g -c camera_capture.c
In file included from camera_capture.c:53:
/usr/include/linux/videodev2.h:2320:20: error: field ‘timestamp’ has
incomplete type
 2320 |  struct timespec   timestamp;
      |                    ^~~~~~~~~

which I wasn't able to solve.  I found this thread about it, which wasn't
too encouraging: https://lkml.org/lkml/2019/10/7/919

The current linux kernel source is slightly different in this area, but I
don't know if it makes a difference
https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/include/uapi/linux/videodev2.h#n2495

-- steve

On Mon, Apr 10, 2023 at 10:57 AM Juan Andres ***@***.***>
wrote:

> Hi,
>
> I found your web camera code to be quite helpful. I forked your repo
> awhile ago, and made some changes to separate the camera code from the SDL
> code, and make like a little camera library.
>
> I also changed the formatting (use tabs, not space, more like linux kernel
> code) and added a Makefile and got rid of the build shell script.
>
> If you would like to merge these changes, feel free, but do not feel any
> obligation to even look at this (I realize you might not like all the
> changes I've made, and also that you haven't touched this code in a long
> time and might have lost interest.)
>
> Mostly just wanted to say thanks, I found your code to be quite helpful.
>
> -- steve
>
> Hi steve, I just saw your message, I don't know why GitHub don't tell me
> about it. I will check your changes, thanks for your contributions
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/djtrance/webcam_viewer/pull/1#issuecomment-1501917442>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAAF22A5MY7FA55LRJ2WC7TXAQNT7ANCNFSM6AAAAAASXUL4JA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

smcameron commented:
Oh yeah, I remember I wrote a bug about this when I discovered it:
https://github.com/smcameron/webcam_viewer/issues/1

On Mon, Apr 10, 2023 at 11:50 AM Stephen Cameron ***@***.***>
wrote:

> No problem.
>
> BTW, on my older linux distro (I forget which one, some flavor of Ubuntu)
> this all worked fine, but lately (linux Mint 20.3 based on Ubuntu focal), I
> get this (which I believe is not a problem with your software, but is
> rather a problem with the linux kernel header files):
>
> $ make
> cc -Wall -Wextra --pedantic -std=c99 -g -c camera_capture.c
> In file included from camera_capture.c:53:
> /usr/include/linux/videodev2.h:2320:20: error: field ‘timestamp’ has
> incomplete type
>  2320 |  struct timespec   timestamp;
>       |                    ^~~~~~~~~
>
> which I wasn't able to solve.  I found this thread about it, which wasn't
> too encouraging: https://lkml.org/lkml/2019/10/7/919
>
> The current linux kernel source is slightly different in this area, but I
> don't know if it makes a difference
>
> https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/include/uapi/linux/videodev2.h#n2495
>
> -- steve
>
> On Mon, Apr 10, 2023 at 10:57 AM Juan Andres ***@***.***>
> wrote:
>
>> Hi,
>>
>> I found your web camera code to be quite helpful. I forked your repo
>> awhile ago, and made some changes to separate the camera code from the SDL
>> code, and make like a little camera library.
>>
>> I also changed the formatting (use tabs, not space, more like linux
>> kernel code) and added a Makefile and got rid of the build shell script.
>>
>> If you would like to merge these changes, feel free, but do not feel any
>> obligation to even look at this (I realize you might not like all the
>> changes I've made, and also that you haven't touched this code in a long
>> time and might have lost interest.)
>>
>> Mostly just wanted to say thanks, I found your code to be quite helpful.
>>
>> -- steve
>>
>> Hi steve, I just saw your message, I don't know why GitHub don't tell me
>> about it. I will check your changes, thanks for your contributions
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/djtrance/webcam_viewer/pull/1#issuecomment-1501917442>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AAAF22A5MY7FA55LRJ2WC7TXAQNT7ANCNFSM6AAAAAASXUL4JA>
>> .
>> You are receiving this because you authored the thread.Message ID:
>> ***@***.***>
>>
>

djtrance commented:
i don't have the last ubuntu or other new version of linux, so I can't check it yet, but I will install a new one to check it. maybe you can find in the linux repo how they are defined  it.
